clang/unittests/StaticAnalyzer/MemRegionDescriptiveNameTest.cpp

```diff
@@ -143,4 +142,18 @@ void top() {
   EXPECT_EQ(Output, "DescriptiveNameChecker: array[x]\n");
 }
 
+TEST(MemRegionDescriptiveNameTest, FieldRegWithSuperElementReg) {
+  StringRef Code = R"cpp(
+void reportDescriptiveName(int *p);
+struct val_struct { int val; };
+extern struct val_struct val_struct_array[3];
```

Comment:
Could you test multidimensional arrays too?
```
extern struct val_struct val_struct_array[3][4];
reportDescriptiveName(&val_struct_array[1][2].val);
```

Reply:
Thanks for the hint. I added a corresponding test case.